tests/unit/mainwindow/test_prompt.py

```diff
@@ -130,3 +131,54 @@ def test_root_path(self, get_prompt):
         """With / as path, show root contents."""
         prompt = get_prompt('/')
         assert prompt._file_model.rootPath() == '/'
+
+    @pytest.fixture
+    def home_path(self, monkeypatch, tmp_path):
+        homedir = str(tmp_path)
+        monkeypatch.setenv('HOME', homedir)  # POSIX
+        monkeypatch.setenv('USERPROFILE', homedir)  # Windows
+        assert str(pathlib.Path.home()) == homedir
+        return tmp_path
+
+    def test_expand_user(self, qtbot, get_prompt, home_path):
+        for directory in ['bar', 'bat', 'foo']:
+            (home_path / directory).mkdir()
+
+        prompt = get_prompt('/')
+        prompt._lineedit.setText('')
+        with qtbot.waitSignal(prompt._file_model.directoryLoaded):
```

Comment:
My bad - pytest-qt has both snake-case (`wait_signal`) and camel-case (`waitSignal`) APIs, and in qutebrowser we decided to consistently use the snake case variant, i.e. `wait_signal`.

Note this doesn't apply to `keyPress` below, because that's API coming 1:1 from Qt.

(same for the other tests)